MusicApp.SongService.Infrastructure/Repositories/ArtistRepository.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.EntityFrameworkCore;
+using MusicApp.SongService.Application.Repositories;
+using MusicApp.SongService.Domain.Entities;
+using MusicApp.SongService.Infrastructure.Data;
+
+namespace MusicApp.SongService.Infrastructure.Repositories;
+
+public class ArtistRepository : IArtistRepository
```

Comment:
maybe you should implement generic base repository and extract all common functionality to it to reduce code duplication